batch/scripts/report_status.sh

```diff
@@ -12,40 +12,38 @@ URL=${STATUS_URL}/${AWS_BATCH_JOB_ID}
 echo "URL: $URL"
 
 # Execute command, save the exit code and output (stdout AND stderr)
-OUTPUT=$("$@" 2>&1)
-EXIT_CODE=$?
+"$@" 2>&1 | tee output.txt
+EXIT_CODE="${PIPESTATUS[0]}"
 
 echo COMMAND EXIT CODE: $EXIT_CODE
-echo OUTPUT: "$OUTPUT"
-
-echo "$OUTPUT" | grep -i error
-GREP_EXIT_CODE=$?
-
-echo GREP EXIT CODE: $GREP_EXIT_CODE
 
 # escape all quotes inside command to not break JSON payload
 # Also make sure we don't reveal any sensitive information
 # But we still want to know if the var was set
 ESC_COMMAND=$(json_escape "$*")
 
-ESC_OUTPUT=$(echo "$ESC_OUTPUT" | sed 's/^AWS_SECRET_ACCESS_KEY.*$/AWS_SECRET_ACCESS_KEY=\*\*\*/') # pragma: allowlist secret
-ESC_OUTPUT=$(echo "$ESC_OUTPUT" | sed 's/^AWS_ACCESS_KEY_ID.*$/AWS_ACCESS_KEY_ID=\*\*\*/')
-ESC_OUTPUT=$(echo "$ESC_OUTPUT" | sed 's/^PGPASSWORD.*$/PGPASSWORD=\*\*\*/')  # pragma: allowlist secret
-ESC_OUTPUT=$(echo "$ESC_OUTPUT" | sed 's/^PGUSER.*$/PGUSER=\*\*\*/')
-ESC_OUTPUT=$(echo "$ESC_OUTPUT" | sed 's/^PGDATABASE.*$/PGDATABASE=\*\*\*/')
-ESC_OUTPUT=$(echo "$ESC_OUTPUT" | sed 's/^PGHOST.*$/PGHOST=\*\*\*/')
-ESC_OUTPUT=$(echo "$ESC_OUTPUT" | sed 's/^SERVICE_ACCOUNT_TOKEN.*$/SERVICE_ACCOUNT_TOKEN=\*\*\*/')
-ESC_OUTPUT=$(echo "$ESC_OUTPUT" | sed 's/^GPG_KEY.*$/GPG_KEY=\*\*\*/')
-ESC_OUTPUT=$(json_escape "$ESC_OUTPUT")
-
-if [ $EXIT_CODE -eq 0 ] && [ $GREP_EXIT_CODE -ne 0 ]; then
+cat output.txt \
+  | sed 's/^AWS_SECRET_ACCESS_KEY.*$/AWS_SECRET_ACCESS_KEY=\*\*\*/' \
+  | sed 's/^AWS_ACCESS_KEY_ID.*$/AWS_ACCESS_KEY_ID=\*\*\*/' \
+  | sed 's/^PGPASSWORD.*$/PGPASSWORD=\*\*\*/' \
+  | sed 's/^PGUSER.*$/PGUSER=\*\*\*/' \
+  | sed 's/^PGDATABASE.*$/PGDATABASE=\*\*\*/' \
+  | sed 's/^PGHOST.*$/PGHOST=\*\*\*/' \
+  | sed 's/^SERVICE_ACCOUNT_TOKEN.*$/SERVICE_ACCOUNT_TOKEN=\*\*\*/' \
+  | sed 's/^GPG_KEY.*$/GPG_KEY=\*\*\*/' \
+  | tail -c 1000 \
+  > filtered_output.txt
+
+ESC_OUTPUT="$(cat filtered_output.txt | python -c 'import json,sys; print(json.dumps(sys.stdin.read())[1:-1])')"
```

Comment:
Can you try to reuse json_escape() function?

Reply:
Done!